src/parsers/maszyna_parser.cpp

```diff
@@ -167,15 +164,10 @@ namespace godot {
 
     Array MaszynaParser::parse() {
         Array result;
-
         while (!eof_reached()) {
-            String token = next_token();
-
-            if (handlers.has(token)) {
-                Callable callback = handlers[token];
-                if (callback.is_valid()) {
-                    Array parsed = callback.call();
-                    if (parsed.size() > 0) {
+            if (String token = next_token(); handlers.has(token)) {
```

Comment:
hard to read code

Reply:
Why? On stream you agreed to that. It is just `if-init` statement
In `if` there's initialized variable only for it's scope, then there's rest after `;` in the statement there.
Nothing hard is there :)

That way we reduce the potential issue of reusing variable where it shouldn't be used

Reply:
> Why? On stream you agreed to that. 
I haven't seen this code.

> That way we reduce the potential issue of reusing variable where it shouldn't be used

And I agreed to this. 
Give me a while to play with this... this is quite strange syntax (for me)